### PR TITLE
Adding 3 new Prefab Instantiation tests

### PR DESCRIPTION
**Pull request by jckand-amzn**
## What does this PR do?

Adds the following automated tests:

test_InstantiatePrefab_LevelPrefab
test_InstantiatePrefab_WithNestedEntities
test_InstantiatePrefab_WithNestedEntitiesAndNestedPrefabs

Also updates an existing prefab to remove the SelectionComponent warning.

## How was this PR tested?

Ran updated Periodic test suite with all tests successfully passing.

### CODE REVIEW

File: AutomatedTesting/Gem/PythonTests/Prefab/tests/instantiate_prefab/InstantiatePrefab_WithNestedEntities.py [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+    import azlmbr.math as math
+
+    from editor_python_test_tools.prefab_utils import Prefab
+    from editor_python_test_tools.wait_utils import PrefabWaiter
+    from editor_python_test_tools.editor_entity_utils import EditorEntity
+    import Prefab.tests.PrefabTestUtils as prefab_test_utils
+
+    NESTED_ENTITIES_PREFAB_FILE_NAME = Path(__file__).stem + 'nested_entities_prefab'
+    NESTED_ENTITIES_NAME_PREFIX = 'Entity_'
+    POSITION = math.Vector3(100.0, 100.0, 100.0)
```
Comment (smurly) [inline AutomatedTesting/Gem/PythonTests/Prefab/tests/instantiate_prefab/InstantiatePrefab_WithNestedEntities.py:31]: this file has import azlmbr.math as math but the next test simply calls azlmbr.math.Vector3 for this line. stylistically consider using one style or the other for consistency

File: AutomatedTesting/Gem/PythonTests/Prefab/tests/instantiate_prefab/InstantiatePrefab_WithNestedEntities.py [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+    NUM_NESTED_ENTITIES_LEVELS = 3
+
+    prefab_test_utils.open_base_tests_level()
+
+    # Creates new nested entities at the root level
+    # Asserts if creation didn't succeed
+    nested_entities_root = prefab_test_utils.create_linear_nested_entities(NESTED_ENTITIES_NAME_PREFIX,
+                                                                           NUM_NESTED_ENTITIES_LEVELS, POSITION)
```
Comment (smurly) [inline AutomatedTesting/Gem/PythonTests/Prefab/tests/instantiate_prefab/InstantiatePrefab_WithNestedEntities.py:39]: stylistically you could newline after the parenthesis in each of these lines to alter the readability, this reduces the dangling parameters on lines by themselves. its really a style trade off choice as long as you consistently go one way or the other it wouldn't matter. spacing in this suggestion may not be accurate to what you would do.
```suggestion
    nested_entities_root = prefab_test_utils.create_linear_nested_entities(
                               NESTED_ENTITIES_NAME_PREFIX, NUM_NESTED_ENTITIES_LEVELS, POSITION)
```

**Review by smurly — APPROVED** [review]

File: AutomatedTesting/Gem/PythonTests/Prefab/tests/instantiate_prefab/InstantiatePrefab_WithNestedEntities.py (hunk 1 above)
Comment (jckand-amzn) [reply AutomatedTesting/Gem/PythonTests/Prefab/tests/instantiate_prefab/InstantiatePrefab_WithNestedEntities.py:31]: Fixed.

File: AutomatedTesting/Gem/PythonTests/Prefab/tests/instantiate_prefab/InstantiatePrefab_WithNestedEntitiesandNestedPrefabs.py [hunk 3]
```diff
@@ -0,0 +1,110 @@
+"""
+Copyright (c) Contributors to the Open 3D Engine Project.
+For complete copyright and license terms please see the LICENSE at the root of this distribution.
+
+SPDX-License-Identifier: Apache-2.0 OR MIT
+"""
+
+
+def InstantiatePrefab_ContainingNestedEntitiesAndNestedPrefabs():
```
Comment (forkercat) [inline AutomatedTesting/Gem/PythonTests/Prefab/tests/instantiate_prefab/InstantiatePrefab_WithNestedEntitiesandNestedPrefabs.py:9]: nit: "Containing" -> "With"

File: AutomatedTesting/Gem/PythonTests/Prefab/tests/instantiate_prefab/InstantiatePrefab_WithNestedEntitiesandNestedPrefabs.py [hunk 4]
```diff
@@ -0,0 +1,110 @@
...
+    nested_entities_root = prefab_test_utils.create_linear_nested_entities(
+        NESTED_ENTITIES_NAME_PREFIX, NUM_NESTED_ENTITIES_LEVELS, CREATION_POSITION)
+    prefab_test_utils.validate_linear_nested_entities(nested_entities_root, NUM_NESTED_ENTITIES_LEVELS,
+                                                      CREATION_POSITION)
+    nested_entities_root_name = nested_entities_root.get_name()
+
+    # Creates new nested prefabs at the root level
+    # Asserts if creation didn't succeed
+    entity_to_nest = EditorEntity.create_editor_entity_at(CREATION_POSITION, name=NESTED_PREFABS_TEST_ENTITY_NAME)
+    assert entity_to_nest.id.IsValid(), "Couldn't create TestEntity"
```
Comment (forkercat) [inline AutomatedTesting/Gem/PythonTests/Prefab/tests/instantiate_prefab/InstantiatePrefab_WithNestedEntitiesandNestedPrefabs.py:53]: nit: f"Couldn't create {NESTED_PREFABS_TEST_ENTITY_NAME}"

**Review by forkercat — APPROVED** [review]

File: AutomatedTesting/Gem/PythonTests/Prefab/tests/instantiate_prefab/InstantiatePrefab_WithNestedEntitiesandNestedPrefabs.py (hunk 4 above)
Comment (jckand-amzn) [reply AutomatedTesting/Gem/PythonTests/Prefab/tests/instantiate_prefab/InstantiatePrefab_WithNestedEntitiesandNestedPrefabs.py]: Fixed.

**Review by cgalvan — APPROVED** [review]

**Pull request merged**